conftest.py

```diff
@@ -32,39 +27,56 @@ def pytest_addoption(parser):
     parser.addoption(
         "--with-answer-testing",
         action="store_true",
-        default=False,
     )
     parser.addoption(
         "--answer-store",
         action="store_true",
-        default=False,
     )
     parser.addoption(
         "--answer-big-data",
         action="store_true",
-        default=False,
     )
     parser.addoption(
-        "--save-answer-arrays",
+        "--answer-raw-arrays",
+        action="store_true",
+    )
+    parser.addoption(
+        "--raw-answer-store",
+        action="store_true",
+    )
+    parser.addoption(
+        "--force-overwrite",
+        action="store_true",
+    )
+    parser.addoption(
+        "--no-hash",
         action="store_true",
     )
+    parser.addoption("--local-dir", default=None, help="Where answers are saved.")
+    # Tell pytest about the local-dir option in the ini files. This
+    # option is used for creating the answer directory on CI
+    parser.addini("local-dir", default="../answer-store", help="answer directory.")
+    parser.addini(
+        "test_data_dir",
+        default=ytcfg.get("yt", "test_data_dir"),
+        help="Directory where data for tests is stored.",
+    )
+    parser.addini(
+        "answer_file_list",
+        default="./tests/tests_pytest.yaml",
+        help="Contains answer file names",
+    )
 
 
 def pytest_configure(config):
     r"""
     Reads in the tests/tests.yaml file. This file contains a list of
     each answer test's answer file (including the changeset number).
     """
-
-    ytcfg["yt", "internals", "within_pytest"] = True
-    # Make sure that the answers dir exists. If not, try to make it
-    if not os.path.isdir(answer_dir):
-        os.mkdir(answer_dir)
-    # Read the list of answer test classes and their associated answer
-    # file
-    with open(answer_file_list) as f:
-        # devnote: this is never used, likely a bug
-        answer_files = yaml.safe_load(f)  # noqa F841
+    yt._called_from_pytest = True
```

Comment:
Is there a specific reason not to use ` ytcfg["yt", "internals", "within_pytest"] = True`?

Reply:
Not that I'm aware of. I just didn't know that was a thing.

Reply:
This may has probably been merged since you forked the repository. I'd suggest using it to prevent using a global variable :)